R/save_e61.R

```diff
@@ -92,162 +120,176 @@ save_e61 <-  function(filename,
+      y_miss <- c(y_miss, i)
+    }
 
-  # For multi-panels: Adjust the width to fit the extra panels and send out user
-  # message to specify the height
-  if (is_multi && is.null(width)) {
-    width <- 8.5 * attr(plot, "panel_cols")
+    # Message if the y-axis label text is too long
+    if (is.null(attr(plots[[i]]$theme, "y_top")) && isTRUE(nchar(plots[[i]]$labels$y) > 5)) {
+      y_long <- c(y_long, i)
+    }
   }
 
-  if (is_multi && is.null(height)) {
-    height <- 7.5 + 7 * (attr(plot, "panel_rows") - 1)
-
-    cli::cli_text(cli::col_green("Note: You are saving a multi-panel graph, save_e61() has automatically set the height to ", height, ", but this value may not be appropriate. Check how the saved graph file looks and adjust the height as required."))
+  # Compile the messages
+  if (length(y_miss) > 0) {
+    y_miss <- paste0(
+      "Plot ",
+      paste(y_miss, collapse = ", "),
+      " is missing a y-axis label. Provide the y-axis units for the reader by specifying the 'y' argument in 'labs_e61()'.")
+  } else {
+    y_miss <- NULL
   }
 
-  # Calculate graph height based on the graph labels for normal orientation graphs
-  if (is.null(height) && !is_flip && !is_multi) {
-
-    h <- 6.5
-
-    # Calculate the height adjustment needed for...
+  if (length(y_long) > 0) {
+    y_long <- paste0(
+      "Plot ",
+      paste(y_long, collapse = ", "),
+      " y-axis labels may be too long. Consider if the information needed to interpret the graph is already in the title and only specify the required units in the y-axis label (for example you could use: %, ppt, $b, '000, n)."
+    )
+  } else {
+    y_long <- NULL
+  }
 
-    # Titles
-    t_adj <-
-      if (!is.null(plot$labels$title)) 0.6 + n_count(plot$labels$title) * 0.3 else 0
+  adv_msg <- c(y_miss, y_long)
 
-    # Subtitles
-    st_adj <-
-      if (!is.null(plot$labels$subtitle)) 0.5 + n_count(plot$labels$subtitle) * 0.3 else 0
+  # Compile and print advisory messages
+  print_adv <- function() {
+    cli::cli_div(theme = list(".bad" = list(color = "#cc0000",
+                                            before = paste0(cli::symbol$cross, " ")),
+                              ".adv" = list(`color` = "#cc0000")
+    )
+    )
+    cli::cli_h1("--- Fix the following issues with your graph ----------------------------------------", class = "adv")
+    cli::cli_ul()
+    sapply(adv_msg, cli::cli_alert, class = "bad")
+    cli::cli_end()
+  }
 
-    # Captions
-    cp_adj <-
-      if (!is.null(plot$labels$caption)) 0.5 + n_count(plot$labels$caption) * 0.3 else 0
+  if (length(adv_msg) > 0) print_adv()
 
-    # Adjustment for width of y-axis label
-    y_adj <-
-      if (!is.null(plot$labels$y)) (nchar(plot$labels$y) - 1) * -0.2 else 0
+  # Require user acknowledgement if there are issues to address
+  # Turn these off in test env
+  if (length(adv_msg) > 0 && !is_testing()) {
```

Comment:
Test to make sure the prompt does not appear if the user:

* Sets y_top = FALSE
* Uses ggplot2::scale_y_anything instead (alt logic, is _not_ using scale_y_continuous_e61, built-in or not)
* Sets a session option to remove the prompt

Reply:
Nice, good thinking!